### PR TITLE
Update JavaLibraryDistributionPlugin

### LINKED ISSUE #5695: JavaLibraryDistributionPlugin seems bugged
### Expected Behavior
- the JavaLibraryDistributionPlugin should inherit from JavaLibraryPlugin (as the name suggests)
- the plugin should pack all runtime dependencies into the tar/zip file

### Current Behavior
- the JavaLibraryDistributionPlugin currently inherits from JavaPlugin (see [here](https://github.com/gradle/gradle/blob/master/subprojects/plugins/src/main/java/org/gradle/api/plugins/JavaLibraryDistributionPlugin.java#L38))
- the plugin tries to pack the dependencies from "runtime" configuration and I think that changed in the past to "runtimeClasspath" (at least this is working for me) (see [here](https://github.com/gradle/gradle/blob/master/subprojects/plugins/src/main/java/org/gradle/api/plugins/JavaLibraryDistributionPlugin.java#L52))

### Context
I want to pack the java library I've created with as less code in the build.gradle file as possible. Currently I need to additionally select the artifacts from the "runtimeClasspath" configuration and I need to add the JavaLibraryPlugin next to the JavaLibraryDistributionPlugin.

### PR DESCRIPTION
**Pull request by GFriedrich**
### Context
Fixes #5695 

### Contributor Checklist
- [X] [Review Contribution Guidelines](https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md)
- [X] Make sure that all commits are [signed off](https://git-scm.com/docs/git-commit#git-commit---signoff) to indicate that you agree to the terms of [Developer Certificate of Origin](https://developercertificate.org/).
- [X] Check ["Allow edit from maintainers" option](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) in pull request so that additional changes can be pushed by Gradle team
- [X] Provide integration tests (under `<subproject>/src/integTest`) to verify changes from a user perspective
- [X] Provide unit tests (under `<subproject>/src/test`) to verify logic
- [x] ~~Update User Guide, DSL Reference, and Javadoc for public-facing changes~~
- [X] Ensure that tests pass locally: `./gradlew <changed-subproject>:check`

### Gradle Core Team Checklist
- [x] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [x] Verify documentation
- [x] Recognize contributor in release notes

### CODE REVIEW

**Comment by GFriedrich** [comment]
@marcphilipp: Is there any chance to have this checked and merged soonish? I would really appreciate it. :sunglasses:

**Comment by marcphilipp** [comment]
@oehme I think this change should be backwards compatible. Any objections?

File: subprojects/plugins/src/integTest/groovy/org/gradle/api/plugins/JavaLibraryDistributionIntegrationTest.groovy [hunk 1]
```diff
@@ -42,6 +42,8 @@ class JavaLibraryDistributionIntegrationTest extends WellBehavedPluginTest {
 
         ${mavenCentralRepository()}
         dependencies {
+            implementation 'commons-io:commons-io:1.4'
+            api 'commons-cli:commons-cli:1.2'
             compile 'commons-collections:commons-collections:3.2.2'
```
[the remark this answers is not in the document]
Comment (GFriedrich) [reply subprojects/plugins/src/integTest/groovy/org/gradle/api/plugins/JavaLibraryDistributionIntegrationTest.groovy]: I left the deprecated configurations intentionally in those tests as I thought they still have to work, but fine by me: Removed "compile" and changed "runtime" to "runtimeOnly".

**Comment by GFriedrich** [comment]
Looking forward to see this in 5.0 +1

**Comment by marcphilipp** [comment]
@GFriedrich Now that we've branched off for the 4.10 release we can continue to work on getting this merged to `master`. Could you please add an entry to the [release notes](https://github.com/gradle/gradle/blob/master/subprojects/docs/src/docs/release/notes.md) mentioning this as a potentially breaking change?

**Comment by GFriedrich** [comment]
@marcphilipp: But isn't there an 4.11 upcoming before a 5.0 is released?

**Comment by GFriedrich** [comment]
@marcphilipp: Added some notes - not sure whether this already fits your needs.

**Comment by marcphilipp** [comment]
Thanks! I polished it a bit and removed the quotes. I think we should really remove those from the template.

@oehme Could you please take another look?

**Comment by marcphilipp** [comment]
@GFriedrich Looks like there was an issue with the commit you branched off of. Could you please rebase onto the current `master`?

**Comment by GFriedrich** [comment]
@marcphilipp done

**Comment by marcphilipp** [comment]
@GFriedrich Thanks! I force pushed to your branch to add the missing sign-off for my commits which is now necessary due to the rebase. And, I started another CI build: https://builds.gradle.org/viewQueued.html?itemId=14927559&tab=queuedBuildOverviewTab

**Comment by marcphilipp** [comment]
@oehme Do you want to take another look at this?

**Comment by marcphilipp** [comment]
Thanks, @GFriedrich! 👍

**Pull request merged**